utils/src/main/scala/za/co/absa/enceladus/utils/validation/SchemaPathValidator.scala

```diff
@@ -136,59 +136,73 @@ object SchemaPathValidator {
     }
   }
 
+  @tailrec
   private def validateSchemaPathArray(schema: StructType,
                                       path: Array[String],
                                       parentOnly: Boolean = false,
                                       fullPathNew: Boolean = false,
                                       parentPath: String = ""): Seq[ValidationIssue] = {
-    // scalastyle:off  return Covered in Issue #730
     if (path.isEmpty) {
-      return Nil
-    }
-
-    val currentField = path(0)
-    val fullPath = s"$parentPath${path.mkString(".")}"
+      Nil
+    } else {
 
-    if (parentOnly && path.length == 1) {
-      if (fullPathNew) {
-        val exactMatch = schema.fields.find(field => field.name == currentField)
-        if (exactMatch.nonEmpty) {
-          return Seq(ValidationError(s"Column '$parentPath$currentField' already exists so it cannot be used as an output column '$fullPath'."))
-        }
-        val caseInsensitiveMatch = schema.fields.find(field => field.name.compareToIgnoreCase(currentField) == 0)
-        if (caseInsensitiveMatch.nonEmpty) {
-          return Seq(ValidationError(s"Case insensitive variant of a cloumn '$parentPath$currentField' already exists so it cannot be used as an output column '$fullPath'."))
+      val currentField = path(0)
```

Comment:
Small: I think the empty lines after else just prolong the function. But if you think it's improve readability keep them (also line 158),